### PR TITLE
Add FILESTRUCTURE.md that describes all the generated files

### LINKED ISSUE #18: Explanation of the file structure in generated README.md
Users might be a bit overwhelmed with the amount of generated files. The generated README should therefore contain a list of which role the files take and under which circumstances you want to change them.

### PR DESCRIPTION
**Pull request by dokempf**
This fixes #18

### CODE REVIEW

**Review by iulusoy — APPROVED** [review]

File: {{cookiecutter.project_slug}}/FILESTRUCTURE.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  a good language for these files, as it is easy to write and rendered into something beautiful by your git repository
+  hosting provider.
+  * `README.md` is the file that users will typically see first when discovering your project.
+  * `COPYING.md` provides a list of copyright holders.
+{%- if cookiecutter.license != "None" %}
+  * `LICENSE.md` contains the license you selected.
+{%- endif %}
+  * `TODO.md` contains a list of TODOs after running the cookiecutter. Following the
+    instructions in that file will give you a fully functional repository with a lot
+    of integration into interesting web services activated and running.
```
Comment (lkeegan) [inline {{cookiecutter.project_slug}}/FILESTRUCTURE.md:89]:
```suggestion
    of integration into useful web services activated and running.
```

File: {{cookiecutter.project_slug}}/FILESTRUCTURE.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+This is an explanation of the file structure that the cookiecutter
+generated for you:
+
+* C++ source files:
+  * `include/{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug }}.hpp` is the main
+    C++ header that declares the interface of your project.
```
Comment (lkeegan) [inline {{cookiecutter.project_slug}}/FILESTRUCTURE.md:6]:
```suggestion
    C++ header that declares the interface of your library.
```

File: {{cookiecutter.project_slug}}/FILESTRUCTURE.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  * `src/{{ cookiecutter.project_slug }}.cpp` is the main file that implements this interface.
```
Comment (lkeegan) [inline {{cookiecutter.project_slug}}/FILESTRUCTURE.md:7]:
```suggestion
  * `src/{{ cookiecutter.project_slug }}.cpp` is the main file that implements this library.
```

File: {{cookiecutter.project_slug}}/FILESTRUCTURE.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    It is built into a library.
```
Comment (lkeegan) [inline {{cookiecutter.project_slug}}/FILESTRUCTURE.md:8]:
```suggestion

```

**Review by lkeegan — APPROVED** [review]
lgtm - feel free to accept/ignore the suggested changes as you prefer

**Pull request merged**